perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -1745,11 +1745,10 @@ public Application rejectApplication(PerunSession sess, int appId, String reason
 	}
 
 	@Override
-	@Transactional(rollbackFor = Exception.class)
 	public void rejectApplications(PerunSession sess, List<Integer> applicationIds, String reason) throws PerunException {
 		Collections.sort(applicationIds, Collections.reverseOrder());
 		for (Integer id : applicationIds) {
-			rejectApplication(sess, id, reason);
+			registrarManager.rejectApplication(sess, id, reason);
```

Comment:
if this is necessary for the calls to be run in separate transactions (?), then the change is required in `approveApplications` as well. Also `approveApplication` is missing the `Transactional` annotation

Reply:
`@Transactional` annotation is on `approveApplicationInternal()` which is called by `approveApplication()`. So there is no transaction on `approveApplication()` itself, since it is kind of wrapper, just like new `approveApplications()`.